ros/content.md

```diff
@@ -58,185 +129,106 @@ Some application may require device access for acquiring images from connected c
 
-## Deployment example
-
-If we want our all ROS nodes to easily talk to each other, we'll can use a virtual network to connect the separate containers. In this short example, we'll create a virtual network, spin up a new container running `roscore` advertised as the `master` service on the new network, then spawn a message publisher and subscriber process as services on the same network.
-
-### Build image
-
-> Build a ROS image that includes ROS tutorials using this `Dockerfile:`
-
-```dockerfile
-FROM %%IMAGE%%:indigo-ros-base
-# install ros tutorials packages
-RUN apt-get update && apt-get install -y \
-    ros-indigo-ros-tutorials \
-    ros-indigo-common-tutorials \
-    && rm -rf /var/lib/apt/lists/
-```
+Alternatively, more permissive network setting can be use to share all host network interfaces with the container, such as [`host` network driver](https://docs.docker.com/network/host/), simplifying connectivity with external network participants. Be aware however that this removes the networking namespace separation between containers, and can affect the ability of DDS participants communicate between containers, as documented [here](https://community.rti.com/kb/how-use-rti-connext-dds-communicate-across-docker-containers-using-host-driver).
 
-> Then to build the image from within the same directory:
-
-```console
-$ docker build --tag %%IMAGE%%:ros-tutorials .
-```
-
-#### Create network
+## Deployment example
 
-> To create a new network `foo`, we use the network command:
+### Docker Compose
 
-	docker network create foo
+In this example we'll demonstrate using [`docker-compose`](https://docs.docker.com/compose/) to spawn a pair of message publisher and subscriber nodes in separate containers connected through shared software defined network.
 
-> Now that we have a network, we can create services. Services advertise there location on the network, making it easy to resolve the location/address of the service specific container. We'll use this make sure our ROS nodes can find and connect to our ROS `master`.
+> Create the directory `~/ros_demos` and add the first `Dockerfile` example from above. In the same directory, also create file `docker-compose.yml` with the following that runs a C++ publisher with a Python subscriber:
 
-#### Run services
+```yaml
+version: '3'
 
-> To create a container for the ROS master and advertise it's service:
+services:
+  talker:
+    build: ./Dockerfile
```

Comment:
```suggestion
    build:
      context: .
      dockerfile: Dockerfile
```

Reply:
This is to avoid errors like:
```
  File "/usr/lib/python3/dist-packages/docker/api/build.py", line 142, in build
    raise TypeError("You must specify a directory to build in path")
TypeError: You must specify a directory to build in path
```

(on docker 19.03.2)

Reply:
Updated examples 66047eb